aws/defaults/handlers.go

```diff
@@ -204,10 +206,25 @@ var RetryMetricHeaderHandler = aws.NamedHandler{
 		if max := r.Retryer.MaxAttempts(); max != 0 {
 			parts = append(parts, fmt.Sprintf("max=%d", max))
 		}
-		// TODO "ttl=YYYYmmddTHHMMSSZ"
-		//    ttl = current_time + socket_read_timeout + estimated_skew
-		// SDK doesn't implement clock skew yet and not obvious how to obtain
-		// read deadlines.
+
+		type timeoutGetter interface {
+			GetTimeout() time.Duration
+		}
+
+		// Attempt extract the TTL from context deadline, or timeout on the client.
+		var ttl time.Time
+		if v, ok := r.Config.HTTPClient.(timeoutGetter); ok && v.GetTimeout() > 0 {
+			ttl = sdk.NowTime().Add(v.GetTimeout())
+		} else if deadline, ok := r.Context().Deadline(); ok {
+			ttl = deadline
+		}
+
+		// Only append the ttl if it can be determined.
+		if !ttl.IsZero() && len(r.AttemptClockSkews) > 0 {
+			const unixTimeFormat = "20060102T150405Z"
+			ttl = ttl.Add(r.AttemptClockSkews[len(r.AttemptClockSkews)-1])
+			parts = append(parts, fmt.Sprintf("ttl=%s", ttl.Format(unixTimeFormat)))
```

Comment:
Will this always be Unix time format for every service?

Reply:
Yeah, per design, this header fields are independent of any API protocols. Generic across all APIs over HTTP.